py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py

```diff
@@ -482,8 +487,207 @@ def pdist(
         raise RuntimeError(
             f"p should between [0, inf], currently p={p} is not supported!"
         )
-    indices = np.triu_indices(shape[0], k=1)
-    return impl.select.index(ctx, target, source_ir, f"{name}_index", norm, indices)
+    if shape[0] == DYNAMIC_DIM:
+        dim = get_shape(ctx, target, source_ir, f"{name}_get_shape", input, 0)
+        shuffle_layer = ctx.net.add_shuffle(dim)
+        shuffle_layer.reshape_dims = trt.Dims()
+        set_layer_name(shuffle_layer, target, f"{name}_shuffle")
+        dim_tensor = shuffle_layer.get_output(0)
+        indices_tensor = tri_upper_indices(
+            ctx, target, source_ir, f"{name}_triu_indices", dim_tensor
+        )
+        gather_layer = ctx.net.add_gather_v2(
+            norm, indices_tensor, mode=trt.GatherMode.ND
+        )
+        set_layer_name(gather_layer, target, f"{name}_gather_layer", source_ir)
+        gather_layer.axis = 2
+        return gather_layer.get_output(0)
+    else:
+        indices = np.triu_indices(shape[0], k=1)
+        return impl.select.index(ctx, target, source_ir, f"{name}_index", norm, indices)
+
+
+def tri_upper_indices(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    size_tensor: TRTTensor,
+) -> TRTTensor:
+    """
+    Return the indices for the upper-triangle part of a square size of matrix in a N-by-2 Tensor,
+    where the diagonal offset = 1. One loop is used to calculate the indices like below.
+    x = 0, y = 0, y_start = 1
+    out_size = size * (size - 1) // 2
+    for _ in range(out_size):
+        y_out.append(y_start + y)
+        x_out.append(x)
+        y += 1
+        if (y_start + y) >= size:
+            x += 1
+            y_start += 1
+            y = 0
+    Args:
+        ctx (ConversionContext): A ConversionContext containing the TensorRT network.
+        target (Target): Target of calling node.
+        source_ir (Optional[SourceIR]): SourceIR of calling converter.
+        name (str): Name of the calling layer.
+        size_tensor (TRTTensor): number of rows in the 2-D square matrix. scalar tensor.
+
+    Example:
+        if size_tensor is 4, it will return [[0, 1], [0, 2], [0, 3], [1, 2], [1, 3], [2, 3]]
+    """
+    zero_layer = ctx.net.add_constant(
+        shape=(), weights=trt.Weights(np.array([0], dtype=np.int32))
+    )
+    set_layer_name(zero_layer, target, f"{name}_zero")
+    constant_0 = zero_layer.get_output(0)
+    one_layer = ctx.net.add_constant(
+        shape=(), weights=trt.Weights(np.array([1], dtype=np.int32))
+    )
+    set_layer_name(one_layer, target, f"{name}_one")
+    constant_1 = one_layer.get_output(0)
+    two_layer = ctx.net.add_constant(
+        shape=(), weights=trt.Weights(np.array([2], dtype=np.int32))
+    )
+    set_layer_name(two_layer, target, f"{name}_two")
+    constant_2 = two_layer.get_output(0)
```

Comment:
Can you directly call `create_constant` in converter_utils.py for simplification?

Reply:
Thanks, It also helped to find wrong import from torch_tensorrt.fx.converters.converter_utils.